SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -744,23 +735,37 @@ void CIncNSSolver::SetTauWall_WF(CGeometry *geometry, CSolver **solver_container
        * In incompressible flows, we can assume that there is no velocity-related
        * temperature change Prandtl: T+ = Pr*y+ ---*/
       su2double T_Wall = nodes->GetTemperature(iPoint);
+      su2double Conductivity_Wall = nodes->GetThermalConductivity(iPoint);
 
-      /*--- Extrapolate the pressure from the interior & compute the
-      wall density using the equation of state ---*/
+      /*--- if a wall temperature was given, we compute the local heat flux using k*dT/dn ---*/
 
+      su2double q_w = 0.0;
+
+      if (config->GetMarker_All_KindBC(iMarker) == ISOTHERMAL) {
+        const su2double T_n = nodes->GetTemperature(Point_Normal);
+        q_w = Conductivity_Wall * (T_Wall - T_n) / WallDistMod;
+      }
+      else if (config->GetMarker_All_KindBC(iMarker) == HEAT_FLUX) {
+        q_w = config->GetWall_HeatFlux(Marker_Tag);  
```

Comment:
divide by reference heat flux
Have you tried a case with non-dimensionalization?